internal/controlplane/handlers_profile.go

```diff
@@ -367,52 +369,102 @@ func (s *Server) getRuleEvaluationStatuses(
 	ruleEvaluationStatuses := make(
 		[]*minderv1.RuleEvaluationStatus, 0, len(dbRuleEvaluationStatuses),
 	)
+	l := zerolog.Ctx(ctx)
+	// Loop through the rule evaluation statuses and convert them to protobuf
 	for _, dbRuleEvalStat := range dbRuleEvaluationStatuses {
-		var guidance string
-
-		// make sure all fields are valid
-		if !dbRuleEvalStat.EvalStatus.Valid ||
-			!dbRuleEvalStat.EvalDetails.Valid ||
-			!dbRuleEvalStat.RemStatus.Valid ||
-			!dbRuleEvalStat.RemDetails.Valid ||
-			!dbRuleEvalStat.EvalLastUpdated.Valid {
-			log.Print("error rule evaluation value not valid")
+		// Get the rule evaluation status
+		st, err := getRuleEvalStatus(ctx, s.store, profileId, dbEntity, selector, providerName, dbRuleEvalStat)
+		if err != nil {
+			l.Err(err).Msg("error getting rule evaluation status")
 			continue
 		}
+		// Append the rule evaluation status to the list
+		ruleEvaluationStatuses = append(ruleEvaluationStatuses, st)
+	}
+	return ruleEvaluationStatuses
+}
 
-		if dbRuleEvalStat.EvalStatus.EvalStatusTypes == db.EvalStatusTypesFailure ||
```

Comment:
@JAORMX wdyt about this, should I revert it? I thought to remove it as it might be needed in the frontend but then again would it be needed in case the rule evaluation is successful?

Reply:
We decided to revert that check 👍